docs/deployment/automated-local.rst

```diff
@@ -0,0 +1,227 @@
+	  --exclude /root/.cache \
+	  --exclude /var/cache \
+	  --exclude /var/lib/docker/devicemapper \
+	  $TARGET::$DATE-$$-system \
+	  / /boot
+
+	borg create $BORG_OPTS \
+	  --exclude 'sh:/home/*/.cache' \
+	  $TARGET::$DATE-$$-system \
+	  /home/
+
+	echo "Completed backup for $DATE"
+
+	# Just to be completely paranoid
+	sync
+
+	if [ -f /etc/backups/autoeject ]; then
+	        umount $MOUNTPOINT
+	        hdparm -Y $drive
+	fi
+
+	if [ -f /etc/backups/backup-suspend ]; then
+	        systemctl suspend
+	fi
+
+Create the ``/etc/backups/autoeject`` file to have the script automatically eject the drive
+after creating the backup. Rename the file to something else (e.g. ``/etc/backup/autoeject-no``)
+when you want to do something with the drive after creating backups (e.g running check).
+
+Create the ``/etc/backups/backup-suspend`` file if the machine should suspend after completing
+the backup. Don't forget to physically disconnect the device before resuming,
+otherwise you'll enter a cycle. You can also add an option to power down instead.
+
+Create an empty ``/etc/backups/backup.disks`` file, you'll register your backup drives
+there.
+
+The last part is to actually enable the udev rules and services:
+
+.. code-block:: bash
+
+	ln -s /etc/backups/40-backup.rules /etc/udev/rules.d/40-backup.rules
+	ln -s /etc/backups/automatic-backup.service /etc/systemd/system/automatic-backup.service
+	systemctl daemon-reload
+	udevadm control --reload
+
+Adding backup hard drives
+-------------------------
+
+Connect your backup hard drive. Format it, if not done already.
+Find the UUID of the file system that backups should be stored on::
+
+    lsblk -o+uuid,label
+
+Note the UUID into the ``/etc/backup/backup.disks`` file.
+
+Mount the drive to /mnt/backup.
+
+Initialize a Borg repository at the location indicated by ``TARGET``::
+
+	borg init --encryption ... /mnt/backup/borg-backups/backup.borg
+
+Unmount and reconnect the drive, or manually start the ``automatic-backup`` service
+to start the first backup::
+
+	systemctl start --no-block automatic-backup
+
+See backup logs using journalctl::
+
+	journalctl -fu automatic-backup [-n number-of-lines]
+
+Security considerations
+-----------------------
+
+The UUID check is a safety / annoyance-reduction mechanism to avoid the script
+blowing up whenever a random USB thumb drive is connected. It is not a security
+mechanism.
+
+Borg ensures that backups are not created on random drives that "just happen"
+to contain a Borg repository. If an unknown unencrypted repository is encountered,
+then the script aborts (BORG_UNKNOWN_UNENCRYPTED_REPO_ACCESS_IS_OK=no).
+
+So backups will only be created on hard drives that have the correct UUID and
+contain a Borg repository that is either known (by ID) to your machine or
+you are using encryption and the passphrases have to match.
+
+Note that generally speaking, physical access to any computer usually equals
+complete compromise of the machine.
```

Comment:
This is increasingly untrue with secure boot, tpm and disk encryption. At least for less advanced attackers.

This does open additional attack surface because it mounts filesystems and scans metadata (of course this is not really avoidable and the uuid check reduces the likelihood without specific targeting). But which could cause parts that are not borg to be attacked (fs drivers and software scanning files on all mounted file systems)